File: PhonoArk/PhonoArk/Models/ExamQuestion.cs
using System.Collections.Generic;

namespace PhonoArk.Models;

public class ExamQuestion
{
    public Phoneme Phoneme { get; set; } = new();
    public List<ExampleWord> Options { get; set; } = new();
    public ExampleWord CorrectAnswer { get; set; } = new();
    public ExampleWord? UserAnswer { get; set; }
    public bool IsAnswered { get; set; }
    public bool IsCorrect => IsAnswered && UserAnswer?.Word == CorrectAnswer.Word;
}

File: pureopus/app/src/main/java/com/phonoark/data/model/ExamQuestion.kt
package com.phonoark.data.model

data class ExamQuestion(
    val phoneme: Phoneme,
    val options: List<ExampleWord>,
    val correctAnswer: ExampleWord,
    var userAnswer: ExampleWord? = null
) {
    val isCorrect: Boolean
        get() = userAnswer?.word == correctAnswer.word

    val isAnswered: Boolean
        get() = userAnswer != null
}
